stdlib/public/Python/NumpyConversion.swift

```diff
@@ -152,41 +152,79 @@ extension Double : NumpyScalarCompatible {
   }
 }
 
-extension Array : ConvertibleFromNumpyArray
-  where Element : NumpyScalarCompatible {
+/// A helper struct with all the data that ConvertibleFromNumpyArray
+/// implementations need to copy memory from a NumpyArray.
+public struct ContiguousNumpyArray<Scalar, ShapeSize> {
```

Comment:
It should be `private`, not `public`, as it's not an API.

Reply:
It's a helper for external modules to use when they implement ConvertibleFromNumpyArray. Concretely, `TensorFlow` uses this.

Reply:
Yes, but it is not an API in that there's no other API than can compose with it. Exposing such an internal data structure is not a Swifty design. In this case, duplicating code should be preferred over exposing an API that's only for internal implementation across modules.